### PR TITLE
Add ICM Filter to the View All Table

### PR DESCRIPTION
**Pull request by okkays**
## What does this change?

- 🌎 Users can currently associate DJ numbers with reports.
- ⛔ They can't later find reports based on those numbers, though
- ✅ This commit adds a DJ number filter, which allows users to filter

Note from issue:

> should we add the same error validation for the sequence number field? (Error message if a non-numerical character is entered in this field)

Rather than add validation, we'll just prevent them from entering it :D

## Screenshots (for front-end PR):

Filter look and feel (plus screenreader):

filter

### Author

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [x] Check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [ ] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.
+ [ ] Re-check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: crt_portal/static/js/complaint_view_filters.js [hunk 1]
```diff
@@ -292,6 +303,26 @@
     }
   };
 
+  /** Produces a new multi-field value given one of its constituent parts. */
+  function buildMultiValue(target, newValue) {
+    const fieldName = target
```
Comment (kewitham) [inline crt_portal/static/js/complaint_view_filters.js:308]: This breaks with per page and grouping because those fields don't have ids

**Review by kewitham — CHANGES_REQUESTED** [review]
just one comment but icm filter functionality looks great!

File: crt_portal/static/js/complaint_view_filters.js (hunk 1 above)
Comment (okkays) [reply crt_portal/static/js/complaint_view_filters.js:308]: I think I've addressed this; can you take another look? (If you're still seeing the issue, can give you me steps to reproduce?)

**Review by ilodidoj — APPROVED** [review]
@okkays passed local docker test.  If @kewitham comments, concerns resolved, this can be merged to dev.

**Review by kewitham — APPROVED** [review]

**Pull request merged**